ompi/mca/io/ompio/io_ompio_file_open.c

```diff
@@ -379,8 +394,9 @@ int mca_io_ompio_file_delete (char *filename,
 
     ret = unlink(filename);
 
-    if (0 > ret) {
-        return OMPI_ERROR;
+    if (0 > ret && ENOENT != errno ) {
+	opal_output (1, "errno = %d %s\n", errno, strerror(errno));
+        return MPI_ERR_ACCESS;
```

Comment:
could this be OMPI_MPI_ERR_ACCESS?  I'm not sure what the Open MPI convention is about what level returns an MPI error code vs an OMPI internal error return value.